internal/services/containers/container_group_resource.go

```diff
@@ -123,6 +121,19 @@ func resourceContainerGroup() *pluginsdk.Resource {
 
 			"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
 
+			"subnets": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &pluginsdk.Schema{
```

Comment:
Can we add  `MaxItems: 1` to this as service team just confirmed there can only be on subnet id set for this at this moment.